Core/include/Acts/EventData/SingleFreeTrackParameters.hpp

```diff
@@ -138,59 +177,15 @@ class SingleFreeTrackParameters {
   /// Momentum three-vector.
   Vector3D momentum() const { return absoluteMomentum() * unitDirection(); }
 
-  /// @brief retrieve electric charge
-  ///
-  /// @return value of electric charge
-  Scalar charge() const { return m_oChargePolicy.getCharge(); }
-
-  /// @brief access to the internally stored FreeParameterSet
-  ///
-  /// @return FreeParameterSet object holding parameter values and their
-  /// covariance matrix
-  const FullFreeParameterSet& getParameterSet() const { return m_oParameters; }
-
-  /// @brief Equality operator
-  ///
-  /// @param [in] rhs Object to compare `*this` to
-  ///
-  /// @return Boolean value whether the objects can be casted into each
-  /// other and the content of the member variables is the same
-  bool operator==(const SingleFreeTrackParameters& rhs) const {
-    auto casted = dynamic_cast<decltype(this)>(&rhs);
-    if (!casted) {
-      return false;
-    }
-
-    return (m_oChargePolicy == casted->m_oChargePolicy &&
-            m_oParameters == casted->m_oParameters);
-  }
-
-  /// @brief inequality operator
-  ///
-  /// @return `not (*this == rhs)`
-  bool operator!=(const SingleFreeTrackParameters& rhs) const {
-    return !(*this == rhs);
-  }
-
-  /// @brief Update of the parameterisation
-  ///
-  /// @param gctx The current geometry context object, e.g. alignment
-  /// @param newValue The new updaed value
-  /// @tparam kIndex Identifier of the parameter index which will be set
-  ///
-  /// @note The context is not used here but makes the API consistent with
-  /// @c SingleCurvilinearTrackParameters and @c SingleBoundTrackParameters
-  template <FreeIndices kIndex>
-  void set(const GeometryContext& /*gctx*/, Scalar newValue) {
-    m_oParameters.setParameter<kIndex>(newValue);
+  /// Particle electric charge.
+  constexpr Scalar charge() const {
+    return m_chargeInterpreter.extractCharge(get<eFreeQOverP>());
   }
 
  private:
-  FullFreeParameterSet
-      m_oParameters;             ///< FreeParameterSet object holding the
-                                 /// parameter values and covariance matrix
-  ChargePolicy m_oChargePolicy;  ///< charge policy object distinguishing
-                                 /// between charged and neutral tracks
+  FullFreeParameterSet m_paramSet;
+  // TODO use [[no_unique_address]] once we switch to C++20
```

Comment:
Can you educate me what this means?

Reply:
Some of the new charge types have no internal state and no member variables. Naively, one would expect that the objects have zero size, but that is not the case. The C++ standard requires concrete instances to have uniquely address; the charge types still occupy the minimal amount of storage (1 byte) to satisfy this requirement. C++20 allows to relax this such that a stateless object actually occupies zero bytes and a pointer to the member variable would in this particular instance point to the memory location after the object proper.

Reply:
Understood - so the proper solution instead of a static variable.